app/models/organization_table_info.rb

```diff
@@ -0,0 +1,165 @@
+class OrganizationTableInfo
+  attr_accessor :organizations_counts, :total_counts
+
+  def initialize(org_stats, total_counts)
+    @organizations_counts = org_stats.kind_of?(Array) ? org_stats : [ org_stats ]
+    @total_counts = total_counts
+  end
+
+  def self.top_stats(top: 0, organization: nil, weeks: 5)
+    # Retrieve the results count, payloads count and organization info for the top most consuming organizations
+    org_payloads_and_results_count = Organization
+      .select('
+        organizations.id, organizations.api_id, organizations.name, organizations.display_name,
+        SUM(test_payloads.results_count) AS test_results_count, COUNT(test_payloads.id) as test_payloads_count
+      ')
+      .joins('
+        LEFT OUTER JOIN "projects" ON "projects"."organization_id" = "organizations"."id"
+        LEFT OUTER JOIN "project_versions" ON "project_versions"."project_id" = "projects"."id"
+        LEFT OUTER JOIN"test_payloads" ON "test_payloads"."project_version_id" = "project_versions"."id"
+      ')
+      .group('organizations.id')
+      .order('test_results_count DESC NULLS LAST')
+
+    # Retrieve the results trends for organizations
+    org_results_trends_count = Organization
+      .select("
+        organizations.id,
+        SUM(test_payloads.results_count) AS test_results_count,
+        date_trunc('week', test_payloads.processed_at) AS trend_date
+      ")
+      .joins('
+        LEFT OUTER JOIN "projects" ON "projects"."organization_id" = "organizations"."id"
+        LEFT OUTER JOIN "project_versions" ON "project_versions"."project_id" = "projects"."id"
+        LEFT OUTER JOIN"test_payloads" ON "test_payloads"."project_version_id" = "project_versions"."id"
+      ')
+      .group("organizations.id, date_trunc('week', test_payloads.processed_at)")
+      .order('organizations.id, trend_date ASC')
+      .where("date_trunc('week', test_payloads.processed_at) >= ?", weeks.weeks.ago.at_beginning_of_week)
+
+    # Add where clause to scope to specific organization
+    if organization
+      org_payloads_and_results_count = org_payloads_and_results_count.where('organizations.id = ?', organization.id)
+      org_results_trends_count = org_results_trends_count.where('organizations.id = ?', organization.id)
+    end
+
+    # Add a limit to do the top n
+    if top > 0
```

Comment:
I don't like having `top` default to 0 and list all organization stats. It's like having a paginated resource that returns all results with no pagination if you don't pass parameters. I would set the default to 5 like we're using now, and I would not allow passing 0 to return all the stats (just like we do with pagination).